module/VuFind/src/VuFind/UrlShortener/Database.php

```diff
@@ -93,19 +106,22 @@ class Database implements UrlShortenerInterface
     /**
      * Constructor
      *
-     * @param string          $baseUrl       Base URL of current VuFind site
-     * @param ShortlinksTable $table         Shortlinks database table
-     * @param string          $salt          HMacKey from config
-     * @param string          $hashAlgorithm Hash algorithm to use
+     * @param string              $baseUrl       Base URL of current VuFind site
+     * @param EntityManager       $entityManager Doctrine ORM entity manager
+     * @param EntityPluginManager $pluginManager VuFind entity plugin manager
+     * @param string              $salt          HMacKey from config
+     * @param string              $hashAlgorithm Hash algorithm to use
      */
     public function __construct(
         string $baseUrl,
-        ShortlinksTable $table,
+        EntityManager $entityManager,
```

Comment:
I would recommend creating a VuFind\Db\Service\ShortLinksService and injecting that, instead of directly interacting with the EntityManager and EntityPluginManager here. Not only does this reduce the number of direct dependencies for this class, but it also makes it easier to ensure that we support custom entity classes. Right now, the logic here is hard-coded to use \VuFind\Db\Entity\ShortLinks, but users should be able to override that using the EntityPluginManager. In order to make that work, we need to use the "getEntityClass" pattern. I think that's easiest if we only create entities through a service class, following the pattern seen in \VuFind\Db\Service\Comments::createEntity (for example).